### PR TITLE
Using Spatie middleware

### PR DESCRIPTION
**Pull request by lloricode**
### Enhancement 
This PR will use a Spatie Middleware
this is the new updates of Spatie https://github.com/spatie/laravel-permission/pull/526

Thanks Best regards :)

### CODE REVIEW

**Comment by lloricode** [comment]
the new Update on Spatie has merge, but we will waiting to update the packagist

**Comment by rappasoft** [comment]
When does this go into effect?

**Comment by lloricode** [comment]
@rappasoft when trying access a admin without any permision or role

**Comment by rappasoft** [comment]
Should it hit the spatie middleware and realize they don't have the view-backend permission and redirect back?

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -52,6 +52,12 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
+        if ($exception instanceof \Spatie\Permission\Exceptions\UnauthorizedException) {
```
Comment (lloricode) [inline app/Exceptions/Handler.php]: @rappasoft it depend on a developer, this is where you can change behavior when `\Spatie\Permission\Exceptions\UnauthorizedException` occured, 

the `UnauthorizedException` also return 403, it already extended with `HttpException`.

**Comment by lloricode** [comment]
btw the packagist of spatie permission is not already updated, to test it, just set `dev-master` on composer.json

**Comment by rappasoft** [comment]
I pulled the code, that exception doesn't exist in their master yet.

**Comment by lloricode** [comment]
it working on with with:
`
screenshot from 2017-10-22 11-17-57

then when try to access url `/admin/dashboard`:
screenshot from 2017-10-22 11-16-57

then the `UnauthorizedException` of spatie on master

**Comment by rappasoft** [comment]
How long until it's master?

**Comment by lloricode** [comment]
We dont know, but we can wait before merge this :)

**Comment by lloricode** [comment]
@rappasoft it already done, we can now update the composer

**Pull request merged**